src/textual/_worker_manager.py

```diff
@@ -0,0 +1,164 @@
+
+    def __iter__(self) -> Iterator[Worker[Any]]:
+        return iter(sorted(self._workers, key=attrgetter("_created_time")))
+
+    def __reversed__(self) -> Iterator[Worker[Any]]:
+        return iter(
+            sorted(self._workers, key=attrgetter("_created_time"), reverse=True)
+        )
+
+    def __bool__(self) -> bool:
+        return bool(self._workers)
+
+    def __len__(self) -> int:
+        return len(self._workers)
+
+    def __contains__(self, worker: object) -> bool:
+        return worker in self._workers
+
+    def add_worker(
+        self, worker: Worker, start: bool = True, exclusive: bool = True
+    ) -> None:
+        """Add a new worker.
+
+        Args:
+            worker: A Worker instance.
+            start: Start the worker if True, otherwise the worker must be started manually.
+            exclusive: Cancel all workers in the same group as `worker`.
+        """
+        if exclusive and worker.group:
+            self.cancel_group(worker.node, worker.group)
+        self._workers.add(worker)
+        if start:
+            worker._start(self._app, self._remove_worker)
+
+    def _new_worker(
+        self,
+        work: WorkType,
+        node: DOMNode,
+        *,
+        name: str | None = "",
+        group: str = "default",
+        description: str = "",
+        exit_on_error: bool = True,
+        start: bool = True,
+        exclusive: bool = False,
+    ) -> Worker:
+        """Create a worker from a function, coroutine, or awaitable.
+
+        Args:
+            work: A callable, a coroutine, or other awaitable.
+            name: A name to identify the worker.
+            group: The worker group.
+            description: A description of the worker.
+            exit_on_error: Exit the app if the worker raises an error. Set to `False` to suppress exceptions.
+            start: Automatically start the worker.
+            exclusive: Cancel all workers in the same group.
+
+        Returns:
+            A Worker instance.
+        """
+        worker: Worker[Any] = Worker(
+            node,
+            work,
+            name=name or getattr(work, "__name__", "") or "",
+            group=group,
+            description=description or repr(work),
+            exit_on_error=exit_on_error,
+        )
+        self.add_worker(worker, start=start, exclusive=exclusive)
+        return worker
+
+    def _remove_worker(self, worker: Worker) -> None:
+        """Remove a worker from the manager.
+
+        Args:
+            worker: A Worker instance.
+        """
+        self._workers.discard(worker)
+
+    def start_all(self) -> None:
+        """Start all the workers."""
+        for worker in self._workers:
+            worker._start(self._app, self._remove_worker)
+
+    def cancel_all(self) -> None:
+        """Cancel all workers."""
+        for worker in self._workers:
+            worker.cancel()
+
+    def cancel_group(self, node: DOMNode, group: str) -> list[Worker]:
```

Comment:
It confused me seeing `node` up here, especially after finding the `cancel_node` method below.
Why don't we just accept the group here? (Maybe with an _optional_ node?)

Reply:
There may be unrelated nodes that share the `group`.

Reply:
Exactly. What if I want to cancel all workers in a single group across multiple nodes?

Reply:
It doesn’t make sense to share groups. I can’t set a use case. Shared groups could also potentially introduce clashes if a third party library happens to have used the same name.

Reply:
Understood.